google/cloud/log.h

```diff
@@ -0,0 +1,399 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_LOG_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_LOG_H_
+/**
+ * @file log.h
```

Comment:
Is this meant to be a simpler (and smaller) version of https://github.com/google/glog ?

What happens if a project wants to use both that logging system and this C++ client library: will they conflict or work well together?

Reply:
Well, I did not know we had open sourced glog :blush:

These little library has some advantages:

- These functions are designed to work with whatever logging library the application developers have picked.  They can inject a backend that capture the logs generated here and sends the output to their log mechanism (including glog, or Boost.Log, or whatever else).

- The cost of run-time disabled logs is extremely low.  No objects are created to discard log lines, glog creates a LogMessage (though it might be fairly cheap).

Naturally glog has many advantages:it has a lot more functionality and mileage than this does.

Reply:
Oh, and I expect one should be able to configure this library to use glog as a backend.